filebeat/beater/filebeat.go

```diff
@@ -56,46 +55,37 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 
 	var err error
 
-	// Init channels
-	fb.publisherChan = make(chan []*input.FileEvent, 1)
+	// Channel from harvesters to spooler
+	publisherChan := make(chan []*input.FileEvent, 1)
 
 	// Setup registrar to persist state
-	fb.registrar, err = crawler.NewRegistrar(fb.FbConfig.Filebeat.RegistryFile)
+	registrar, err := registrar.NewRegistrar(fb.FbConfig.Filebeat.RegistryFile)
```

Comment:
I like the registrar finally being a separate packet. A more golang idiomatic way of initializing the registrar is `registrar.New(...)`.